NETDefault/UnitTestingWithNUnit&Moq/ServiceTests.cs

```diff
@@ -0,0 +1,74 @@
+using Moq;
+using NUnit.Framework;
+
+namespace ServiceTests
+{
+    internal class ServiceTests : ServiceTestsBase
+    {
+        [Test]
+        public void GetDashboard_WhenCustomerNameNotFound_ItShouldThrow()
+        {
+            // Arrange
+            SetupGetUserIdByUserName(unknownUsername);
+
+            //Assert
+            Assert.Throws<NotFoundException>(() => _subject.GetDashboardByUserName(unknownUsername));
+
+            _gateway.Verify(x => x.GetUserIdByUserName(unknownUsername), Times.Once());
+            _gateway.VerifyNoOtherCalls();
+        }
+
+        [Test]
+        public void GetDashboard_WhenCustomerNameFound_ItShouldBeReturned()
+        {
+            // Arrange
+            SetupGetUserIdByUserName(knownUsername);
+
+            var expected = userId.ToString();
+
+            //Act
+            var result = _subject.GetDashboardByUserName(knownUsername);
+
+            //Assert
+            Assert.AreEqual(expected, result);
+
+            _gateway.Verify(x => x.GetUserIdByUserName(knownUsername), Times.Once());
+            _gateway.Verify(x => x.GetHobbiesByUserId(userId), Times.Once());
+        }
+
+        [Test]
+        public void GetDashboard_WhenUserFound_ItShouldGetHobbies()
+        {
+            // Arrange
+            SetupGetUserIdByUserName(knownUsername);
+            SetupGetHobbiesByUserId(userId);
+
+            //Act
+            var result = _subject.GetDashboardByUserName(knownUsername);
+
+            //Assert
+            _gateway.Verify(x => x.GetUserIdByUserName(knownUsername), Times.Once());
+            _gateway.Verify(x => x.GetHobbiesByUserId(userId), Times.Once());
+            _gateway.VerifyNoOtherCalls();
+        }
+
+        [Test]
```

Comment:
I would change the test here to incorporate some `[TestCase]`s or a `[TestCaseSource]` so the example also demonstrates that too.

Showing how a general tests can be made with 4 or 5 different inputs is very valuable. It shows that you don't have to copy paste the same test over and over again if you want to test different scenarios.